### ISSUE TITLE
Compact explorer tree nodes don't narrate "Expanded" and "Collapsed"

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#86414]
Using NVDA to test https://github.com/microsoft/vscode/issues/85039: clicking on any of the compact items expands and collapses the node. However, it doesn't read "Expanded" and "Collapsed" the way non-compact nodes do.
Reads "Expanded" and "Collapsed" when I expand and collapse, respectively.
![noncompactnode](https://user-images.githubusercontent.com/38270282/70250976-09261e80-177f-11ea-838d-a660d1834efc.gif)
Does not read "Expanded" and "Collapsed" when I expand and collapse, respectively.
![compactnode](https://user-images.githubusercontent.com/38270282/70250974-09261e80-177f-11ea-9872-2b90880eb110.gif)

**Comment by isidorn** [comment]
Good feedback. Thanks. Assigning to december to investigate.

**Comment by isidorn** [comment]
We do not set all the needed aria properties on the subnodes. One of them being the expanded and collapsed state.
So idealy we would set the `aria-expanded` somewhere [here](https://github.com/Microsoft/vscode/blob/5c569d1cebbbe4ffad74c6cf76111f48b53fb316/src/vs/workbench/contrib/files/browser/views/explorerViewer.ts#L164)
Missing issue: currently the navigationController has no event to listen to update the expanded flag.
I do not think it is a big issue, thus assigning to backlog candidates until users complain.

**Comment by davidackroyd99** [comment]
Hi, please could I have this issue? Saw [this](https://www.microsoft.com/en-gb/business/azure/oss/?wt.mc_id=AID3003006_QSG_PD_SCL_325&OCID=AID3003006_OLA_23856221_268787623_129304026) ad on Reddit and thought I should get involved!

**Comment by isidorn** [comment]
@davidackroyd99 this issue is not so trivial, however if you would like to take a stab feel free. Once you have something working ping me on the PR. How the tree sets the activedescendent for the compact subnodes might be a good example how this could potentialy be done.

**Comment by davidackroyd99** [comment]
cheers :) will let you know when I have something!

**Comment by Neurrone** [comment]
Pending a proper fix, could `explorer.compactFolders` default to false if not otherwise set if screen reader usage is detected? It causes some difficulty when trying to navigate in the explorer view.

**Comment by isidorn** [comment]
@Neurrone let's assign to next milestone so we look into a proper fix. Thanks for bringing this up.
If we do not fix it, we can disable the compactFolders for screen readers.

**Comment by isidorn** [comment]
Pushed a fix for this such that we now properly set the `aria-expanded` and `aria-level` property on the compact nodes.
@Neurrone try it out and let us know what you think. Thanks!

**Comment by alexr00** [comment]
There's still something weird happening here. I get an extra "collapsed" when I click on the rightmost folder to expand it. I just expanded it, should it read collapsed like that (it doesn't for non-compact nodes).
![recording (6)](https://user-images.githubusercontent.com/38270282/86349233-1649b400-bc61-11ea-8635-9197d2d92f67.gif)
narrates:
level 6  classification  not selected  collapsed
expanded

**Comment by isidorn** [comment]
Thanks @alexr00 for looking into this. I just checked the corner case you mentioned and I can indeed reproduce this. However since screen readers users do not use the mouse, and this issue can not be reproduced with the keyboard I am fine living with this.
Thus closing this and adding verified label as you said that it can be considered verified.
